### DISCUSSION TITLE
v3.9.3

### DISCUSSION THREAD

**Discussion by jsjoeio** [coder/code-server#3075] [General]
VS Code v1.54.2

Upgrading is as easy as installing the new version over the old one. code-server
maintains all user data in `~/.local/share/code-server` so that it is preserved in between
installations.

## New Features

⭐ Changes to CentOS support (see [docs/install.md](https://github.com/cdr/code-server/blob/main/docs/install.md#fedora-centos-rhel-suse) for more info)

  - dropped CentOS 7 support on official arm64 builds #2966 

## Bug Fixes

⭐ Fix bug caused in 3.9.2 on Safari/iOS causing blank screen

  - Don't send permessage-deflate header if not supported #2993 

## Documentation

⭐ A couple minor updates to the docs.

  - docs(install): add Termux workaround for search #3039 
  - chore: fix typo in install.sh #3025 (thank you @eltociear 🎉)

## Development

⭐ Faster CI, security updates, improved release workflow and extend use of Ranger GitHub bot for issues

  - completely refactored CI to be more stable and faster #2966 (huge thanks to @oxy 🎉)
  - fix: update ranger file #3065 
  - dev(ranger): add action for upstream issues #3026 
  - dev(ranger): add action for extension request #2967 
  - dev(testing): add jest-playwright and reduce flakiness of e2e tests #3016 
  - feat(ci): add audit job for security #3042 (inspired by @PatrickDerichs 💛)
  - fix(deps): update and fix vulnerabilities #3041 (thank you @PatrickDerichs for flagging 🎉)
  - Fix tsc watch continually restarting during vscode build #3015 
  - feat(ci): add homebrew job to publish workflow #3010 
  - fix(dev-deps): update ts-node and spawn #2999 
  - refactor: reorganize release notes draft format #2969 
  - refactor: add check for artifacts url in lib.sh #2968 

Cheers! 🍻

<hr /><em>This discussion was created from the release <a href='https://github.com/cdr/code-server/releases/tag/v3.9.3'>v3.9.3</a>.</em>

**Comment by jsjoeio** [comment]
NOTE: the publish workflow [failed](https://github.com/cdr/code-server/runs/2300619746?check_suite_focus=true#step:3:1110) to download `nfpm` and we are working on fixing this to get 3.9.3 published on npm and Docker. 

See PR: https://github.com/cdr/code-server/pull/3076

**Reply by jsjoeio** [reply]
Fixed! See [run](https://github.com/cdr/code-server/runs/2300894712?check_suite_focus=true).

[Published on npm](https://www.npmjs.com/package/code-server)

[Published on Docker hub](https://hub.docker.com/r/codercom/code-server/tags?page=1&ordering=last_updated)

[Published AUR package](https://aur.archlinux.org/packages/code-server/)

[Published on homebrew](https://github.com/Homebrew/homebrew-core/pull/74839)

**Comment by raisin8010** [comment]
Why not upgrade to VS Code v1.54.3?

**Reply by oxy** [reply]
We're planning to skip straight to 1.55 in the next release - the work involved to fix merge conflicts isn't really worth it to us when the next major version is out already. This release was mostly to get a stable fix for the Safari issue which landed in 3.9.2.

**Comment by lg-nineoneninethreenine** [comment]
Can I deploy it in Railway, because I use it deployed, can I just go to deploy-code-server, fork it and do all the rest?

**Reply by lg-nineoneninethreenine** [reply]
I mean ,like, how can I update my deployment

**Reply by jsjoeio** [reply]
Yes! deploy-code-server has a guide you can follow [here](https://github.com/cdr/deploy-code-server/blob/main/guides/railway.md). 

To update, change the version in the Dockerfile to `3.9.3` like this:
```Docker
FROM codercom/code-server:3.9.3
```

**Reply by lg-nineoneninethreenine** [reply]
Thanks!😀😀😀

**Comment by lg-nineoneninethreenine** [comment]
One last doubt
How can I change my Bash username, because, I feel I wanna change it to something I like. Should I change it in the Dockerfile?

**Reply by jsjoeio** [reply]
Here's a great answer from @nooobcoder:
https://github.com/cdr/code-server/discussions/3055#discussioncomment-577942

**Reply by code-asher** [reply]
I haven't tried this in a while but it should also be possible to set your user in the container with `DOCKER_USER`: 

Docs: https://github.com/cdr/code-server/blob/e07ae004b73823483e9f7015548dc7d38334558a/docs/install.md#docker

**Reply by PatrickDerichs** [reply]
If you do want to use a Dockerfile, you can use the commands (use the RUN directive) in link in @jsjoeio 's comment. You can then use the USER directive to switch to your user

```
FROM codercom/code-server:3.9.3

RUN sudo adduser <username>
RUN sudo usermod -aG sudo <username>
USER <username>
```

I haven't tried this out myself, but that is an option you could try.

**Reply by nooobcoder** [reply]
> I haven't tried this in a while but it should also be possible to set your user in the container with `DOCKER_USER`:
> 
> Docs: https://github.com/cdr/code-server/blob/e07ae004b73823483e9f7015548dc7d38334558a/docs/install.md#docker

Thanks for this @code-asher this was not known to me, thank you

**Reply by code-asher** [reply]
Of course! I should warn though that it looks like there's a bug with the `DOCKER_USER` method if you start a stopped container: https://github.com/cdr/code-server/issues/2767

**Reply by nooobcoder** [reply]
@coder-lg please have an insight on this, @code-asher has mentioned of a bug in the previous reply. Proceed with caution. The bug seems very serious, so I thought warning you. #2767
